packages/javascript-sdk/src/http-client/index.ts

```diff
@@ -167,20 +172,15 @@ abstract class HttpClient extends Dispatcher {
     return headers;
   }
 
-  private static async stepIterator(
-    res: Response,
-    handleStep: HandleStep,
-    type: string,
-    tree: string,
-  ): Promise<void> {
+  private static async stepIterator(res: Response, handleStep: HandleStep): Promise<void> {
     const jsonRes = await res.json();
     const initialStep = new FRStep(jsonRes);
 
     // eslint-disable-next-line no-async-promise-executor
     return new Promise(async (resolve, reject) => {
       async function handleNext(step: FRStep): Promise<void> {
         const input = await handleStep(step);
-        const output = await FRAuth.next(input, { type, tree });
+        const output = await FRAuth.next(input, { tree: '', type: '' });
```

Comment:
I eventually figured out that in order for the transaction approval tree-outcome to take effect, we could only pass the `authIndexType=composite_advice` on the initial authentication request. Having any query parameters for `authIndexType` or `authIndexValue` after the first one, caused a 401 described as "session expired". It was weird but I found a conversation @cerebrl had with others in the team, where this was discovered. I'm fine having this in here, but we may want to discuss why it is necessary for this flow now when previously that may not have been the case - @cerebrl.

Reply:
Hey David, sorry for not properly documenting this. I remember writing it down somewhere, but I can't, for the life of me, remember where. You are correct. Something in AM has changed that made it much more sensitive to these parameters. If I remember correctly, the fix is removing the `tree` property from the root SDK config. That way, only when a `tree` value is explicitly passed in does it add those parameters.

Reply:
Ok, so are you saying that we do not need this code change - and instead require our tests (and therefore the users of the SDK), to remove the tree property from the root config in order for the tree to be omitted and this flow to work? I'm fine with removing it from our tests but would be more concerned about pushing developers into configuring the SDK a certain way for this to work. I guess if we at least document this then it's ok, but perhaps we should think about raising a ticket on AM or adapting the SDK similar to the above change. What do you think?

Reply:
I'm going to write up a story to improve the core module of the SDK. That way we can track this. As far as this particular use case, you may have to preserve this fix, but I'd like to address the underlying issue, rather than patch this symptom.

Reply:
Ok I think that makes sense. Thanks for creating the story - I'll keep this code as-is for now.